eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -32,59 +42,93 @@ function Get-SwaggerInfo()
             }
-        } elseif ($inputfile -ne "") {
-            Write-Host "Updating autorest.md file to update input-file."
-            $inputfileRex = "input-file *:*"
-            if ((Get-Content $autorestFilePath | Select-String -Pattern $inputfileRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -replace $inputfileRex, ("input-file:" + [Environment]::NewLine + "$inputfile") | Set-Content $autorestFilePath
-            } else {
-                $startNum = (Get-Content $autorestFilePath | Select-String -Pattern '```').LineNumber[0]
-                $fileContent = Get-Content -Path $autorestFilePath
-                $fileContent[$startNum - 1] += ([Environment]::NewLine + "input-file:" + [Environment]::NewLine + "$inputfile")
-                $fileContent | Set-Content $autorestFilePath
+        }
+        
+        # update autorest.md with configuration
+        if ( $autorestConfigYaml) {
+            Write-Host "Update autorest.md with configuration."
+            $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+            if ( $range.count -gt 1) {
+                $startNum = $range[0];
+                $lines = $range[1] - $range[0] - 1
+                $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
             }
-            
-            if ( $? -ne $True) {
-                Write-Error "Failed to update autorest.md. exit code: $?"
-                exit 1
+
+            Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+            Import-Module powershell-yaml
+            $yml = ConvertFrom-YAML $autorestConfigYaml
+
+            $fileContent = Get-Content -Path $autorestFilePath
+            Foreach ( $key in $yml.keys) {
+                if ( ($key -eq "output-folder") -or ($key -eq "require")) {
+                    continue;
+                }
+                $match = ($fileContent | Select-String -Pattern $key).LineNumber
+                if ($match.count -gt 0) {
+                    $fileContent[$match[0] - 1] = $key + ":" + $yml[$key];
+                } else {
+                    $startNum = ($fileContent | Select-String -Pattern '```').LineNumber[0]
+                    $fileContent[$startNum - 1] += ([Environment]::NewLine + $key + ":" + $yml[$key])
+                }
             }
-        }   
+
+            $fileContent | Out-File $autorestFilePath
+        }
     } else {
-        Write-Error "autorest.md doesn't exist."
-        exit 1
+        Write-Host "autorest.md does not exist. start to create one."
+        if ( $autorestConfigYaml ) {
+            Write-Host "Create autorest.md with configuration."
+            $autorestConfigYaml = "# Azure.IoT.DeviceUpdate" + [Environment]::NewLine + '``` yaml' +  [Environment]::NewLine + $autorestConfigYaml + '```' + [Environment]::NewLine;
```

Comment:
Did we intend to hardcode Azure.IoT.DeviceUpdate here?

Reply:
updated